canals/component/component.py

```diff
@@ -194,6 +197,13 @@ def run(self, data: <Input if defined, otherwise untyped>) -> <Output if defined
     class_.__canals_input__ = input_
     class_.__canals_output__ = output
```

Comment:
That's assuming that the user is decorating an `input` or `output` method, but that could be a wrong assumption. We don't enforce naming for those method, they could be named `foo` and `bar` for all we care.

Saving those here makes it easy for us to find the I/O dataclasses instead of searching for them each time.

Reply:
Aaah that's right, I forgot that :+1: